modules/03-network-resiliency/fault-injection/README.md

```diff
@@ -0,0 +1,217 @@
+# Network Resiliency - Fault Injection
+
+This sub-module focuses on Istio service-mesh features like **Delay and Abort** for network resiliency on Amazon EKS. 
+
+1. [Delay](#injecting-delay-fault-into-http-requests)
+2. [Abort](#injecting-abort-fault-into-http-requests)
+
+## Injecting Delay Fault into HTTP Requests
+
+In this step we setup the delay configuration for `catalogdetail` virtual service
+
+### Deploy
+
+```sh
+# This assumes that you are currently in "istio-on-eks/modules/03-network-resiliency/fault-injection" directory
+
+kubectl apply -f ./delay/catalogdetail-virtualservice.yaml
+```
+
+Output should be similar to:
+```sh
+virtualservice.networking.istio.io/catalogdetail configured
+```
+
+### Validate
+
+Run the following command to retrieve the YAML configuration for [`catalogdetail`](./delay/catalogdetail-virtualservice.yaml) `VirtualService`.
+
+```sh
+kubectl get virtualservice catalogdetail -o yaml -n workshop
+```
+
+Output should be similar to:
+```sh
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"networking.istio.io/v1alpha3","kind":"VirtualService","metadata":{"annotations":{},"name":"catalogdetail","namespace":"workshop"},"spec":{"hosts":["catalogdetail"],"http":[{"fault":{"delay":{"fixedDelay":"15s","percentage":{"value":100}}},"match":[{"headers":{"user":{"exact":"internal"}}}],"route":[{"destination":{"host":"catalogdetail","port":{"number":3000}}}]},{"route":[{"destination":{"host":"catalogdetail","port":{"number":3000}}}]}]}}
+  creationTimestamp: "2024-01-19T17:47:48Z"
+  generation: 12
+  name: catalogdetail
+  namespace: workshop
+  resourceVersion: "35171847"
+  uid: 2060b9cd-1e4b-4127-bf94-5c6bc679f286
+spec:
+  hosts:
+  - catalogdetail
+  http:
+  - fault:
+      delay:
+        fixedDelay: 15s
+        percentage:
+          value: 100
+    match:
+    - headers:
+        user:
+          exact: internal
+    route:
+    - destination:
+        host: catalogdetail
+        port:
+          number: 3000
+  - route:
+    - destination:
+        host: catalogdetail
+        port:
+          number: 3000
+```
+
+### Test
+
+Test the delay by running a `curl` command against the `catalogdetail` service from within the mesh for user named 'internal' and 'external'.
+
+```sh  
+# Create a multitool pod to be able to use curl from within the mesh
+
+kubectl run multitool --image=praqma/network-multitool -n workshop
```

Comment:
`praqma/network-multitool` project is moved to [`wbitt/Network-MultiTool`](https://github.com/wbitt/Network-MultiTool).

Change the image to point to `wbitt/network-multitool` instead.